File: src/test/java/com/znsio/teswiz/screen/theapp/AppLaunchScreen.java
package com.znsio.teswiz.screen.theapp;


public abstract class AppLaunchScreen {

    public static AppLaunchScreen get() {
        return com.znsio.teswiz.screen.ScreenRegistry.getScreen(AppLaunchScreen.class);
    }

    public abstract LoginScreen selectLogin();

    public abstract AppLaunchScreen goBack();

    public abstract EchoScreen selectEcho();

    public abstract ClipboardDemoScreen goToClipboardDemo();
}

File: src/test/resources/playwright/screens/theapp/app-launch.screen.ts
import type { ScreenContext } from "../screen-context.ts";
import { stayOnCurrentScreen, unsupportedScreenAction } from "../screen-route.ts";

const LOCATORS = {
  loginLink: "Form Authentication",
} as const;

export async function selectLogin(screen: ScreenContext): Promise<void> {
  await screen.page.getByRole("link", { name: LOCATORS.loginLink }).click();
}

export function goBack() {
  return stayOnCurrentScreen();
}

export function selectEcho() {
  return unsupportedScreenAction(
    "TheApp Echo screen is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}

export function goToClipboardDemo() {
  return unsupportedScreenAction(
    "TheApp Clipboard Demo screen is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}
